frontend/src/components/edit-profile/edit-profile.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { useEffect, useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import CrossIcon from '@assets/images/edit-profile/cross.svg';
+import DefaultAvatar from '@assets/images/edit-profile/default-avatar.png';
+import PencilIcon from '@assets/images/edit-profile/pencil.svg';
+import TrashIcon from '@assets/images/edit-profile/trash.svg';
+import { updateUserSchema } from '@autoline/shared';
+import { AppRoute } from '@common/enums/enums';
+import { ButtonFill } from '@components/common/button-fill/button-fill';
+import { ButtonOutline } from '@components/common/button-outline/button-outline';
+import { InputField } from '@components/common/input-field/input-field';
+import { DialogDeleteAccount } from '@components/edit-profile/dialog-delete-account/dialog-delete-account';
+import { SelectYearRange } from '@components/edit-profile/select-years-range/select-year-range';
+import { SignIn } from '@components/edit-profile/sign-in/sign-in';
+import { useAppForm } from '@hooks/app-form/app-form.hook';
+import { Alert, MenuItem, Modal, Stack } from '@mui/material';
+import {
+  ProfileFieldsRequestData,
+  useDeleteUserProfileMutation,
+  useUpdateUserProfileMutation,
+} from '@store/queries/user/update-user';
+
+import { SelectFieldForm } from './select-field-form/select-field-form';
+import styles from './styles.module.scss';
+
+interface EditProfileProps {
+  onClose: () => void;
+}
+
+export const EditProfile: React.FC<EditProfileProps> = ({ onClose }) => {
+  const user = {
+    sex: null,
+    birthYear: '1996',
+    location: 'kyiv',
+    name: 'stepan shevchenko',
+    phone: '+380938889922',
+    email: 'stepan1909@gmail.com',
+  };
+  const navigate = useNavigate();
+  const [openDialog, setOpenDialog] = useState(false);
+  const [
+    updateUserProfile,
+    {
+      isLoading: updateIsLoading,
+      isSuccess: updateIsSuccess,
+      error: updateError,
+    },
+  ] = useUpdateUserProfileMutation();
+
+  const [
+    deleteUserProfile,
+    {
+      isLoading: deleteIsLoading,
+      isSuccess: deleteIsSuccess,
+      error: deleteError,
+    },
+  ] = useDeleteUserProfileMutation();
+
+  const { control, errors, handleSubmit, setValue } =
+    useAppForm<ProfileFieldsRequestData>({
+      defaultValues: {
+        sex: user.sex || 'not_appliable',
+        birthYear: user.birthYear || 'not_appliable',
+        location: user.location || 'not_appliable',
+        name: user.name,
+        phone: user.phone,
+        email: user.email,
+      },
+      validationSchema: updateUserSchema,
+    });
+
+  useEffect(() => {
+    if (deleteIsSuccess) {
+      navigate(AppRoute.ROOT);
+    }
+  });
```

Comment:
You didn't use the `useEffect` dependencies. This effect will run on each rerender